lib/datadog/core/configuration/ext.rb

```diff
@@ -19,6 +19,22 @@ module Metrics
 
         module Transport
```

Comment:
I would consider changing this name from `Transport`: this name narrows its purpose unnecessarily to just use with `Transport`, and what its describing doesn't really pertain to `Transport`. I might suggest something like `Agent`/`TraceAgent`/`APMAgent`.

Reply:
I like the idea of using `Agent` as a module name here. But, this file on [master](https://github.com/DataDog/dd-trace-rb/blob/master/lib/datadog/core/configuration/ext.rb) already has `ENV_DEFAULT_HOST` defined under `Datadog::Core::Configuration::Ext::Transport` and it is referenced in the upgrade guide [here](https://github.com/DataDog/dd-trace-rb/blob/master/docs/UpgradeGuide.md?plain=1#L661). Is there any harm in changing this, and updating the doc? 

Alternatively, we could leave `ENV_DEFAULT_HOST` under `Transport` and move the items relevant to my PR into `Agent`, but that does feel a little bit odd.

Reply:
> it is referenced in the upgrade guide [here](https://github.com/DataDog/dd-trace-rb/blob/master/docs/UpgradeGuide.md?plain=1&rgh-link-date=2023-08-23T14%3A31%3A09Z#L661). Is there any harm in changing this, and updating the doc?

Since we directly instructed clients to use `Datadog::Core::Configuration::Ext::Transport::ENV_DEFAULT_HOST`, we shouldn't break this contract in `ddtrace` 1.x.

> Alternatively, we could leave ENV_DEFAULT_HOST under Transport and move the items relevant to my PR into Agent, but that does feel a little bit odd.

I suggest doing this, despite this solution being a compromise.
I had to do something similar for distributed trace headers: https://github.com/DataDog/dd-trace-rb/blob/f3c7f9672ff4a9d6b5fbc6a2de3a276d294a061e/lib/datadog/tracing/distributed/headers/ext.rb#L7
These constants were moved to their respective implementation classes (e.g. https://github.com/DataDog/dd-trace-rb/blob/f3c7f9672ff4a9d6b5fbc6a2de3a276d294a061e/lib/datadog/tracing/distributed/headers/ext.rb#L20 was moved to https://github.com/DataDog/dd-trace-rb/blob/f3c7f9672ff4a9d6b5fbc6a2de3a276d294a061e/lib/datadog/tracing/distributed/b3_multi.rb#L13), but I'm not able to completely remove them from their original location because they are very much public.

As long as we have a comment documentation that we should remove these constants on the next major release, it's ok to have them duplicated for the time being.

Reply:
@marcotc That's a good point. I don't think I fully understood that users are directly references the constants like that. I'll move `ENV_DEFAULT_HOST` back under transport.